components/HistoryMessages.tsx

```diff
@@ -2,40 +2,53 @@ import React from "react";
 import { Message } from "@/lib/actions/message.actions";
 import { User } from "@/lib/actions/user.actions";
 
-export default function HistoryMessages({
-  messages,
-  user,
-}: {
+interface HistoryMessagesProps {
   messages: Message[];
   user: User;
-}) {
-  const formatTimestamp = (timestamp) => {
+}
+
+const HistoryMessages: React.FunctionComponent<HistoryMessagesProps> = ({ messages, user }) => {
+  const formatTimestamp = (timestamp: string) => {
     const dateObject = new Date(timestamp);
-    const hours = dateObject.getHours().toString().padStart(2, "0");
-    const minutes = dateObject.getMinutes().toString().padStart(2, "0");
-    return `${hours}:${minutes}`;
+    const currentDate = dateObject.toDateString();
+    const currentTime = dateObject.toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" });
+    return { date: currentDate, time: currentTime };
   };
 
+  let currentDate: string | null = null;
+
   return (
     <div className="w-full h-full flex flex-col items-stretch">
       <div className="bg-teal-500 text-white py-2 text-center">
         <h1 className="text-lg font-bold">All Messages</h1>
       </div>
       <div className="flex-none flex flex-col items-start overflow-y-auto p-4">
-        {messages?.map((message, index) => (
-          <div
-            key={index}
-            className={`relative rounded p-4 mb-4 max-w-content ${
-              message.senderId === user.id ? "bg-teal-200 bg-opacity-75 self-end" : "bg-gray-200 bg-opacity-75 self-start"
-            }`}
-          >
-            <div>{message.messageContent}</div>
-            <div className="text-xs text-gray-500 absolute bottom-1 right-1">
-              {formatTimestamp(message.timeStamp)}
+        {messages?.map((message, index) => {
+          const { date, time } = formatTimestamp(message.timeStamp);
+          const isNewDate = date !== currentDate;
+          currentDate = date;
+
+          return (
+            <div key={index} className="flex items-start w-full justify-center">
+              {isNewDate && (
+                <div className="text-center mb-2 w-full max-w-content">
+                  <strong>{date}</strong>
+                </div>
+              )}
+              <div
+                className={`relative rounded p-4 mb-4 max-w-content min-w-[65px] ${
+                  message.senderId === user.id ? "bg-teal-200 bg-opacity-75 self-end ml-auto" : "bg-gray-200 bg-opacity-75 self-start mr-auto"
+                }`}
+              >
+                <div style={{ whiteSpace: "nowrap" }}>{message.messageContent}</div>
```

Comment:
The "style" tag is fine, but it may be better to follow Tailwind CSS rules : `<div className="whitespace-nowrap">`, 
also it seems like have overlap the message box by text:
![Screenshot (44)](https://github.com/gallery-apps/whats-app2/assets/51173079/646a0471-7d86-492f-8547-6b56f229d01d)
May be switch from "nowrap" to "normal" `<div className="whitespace-normal">`
@HamzaAli303